Sources/SecureStorage/SecureStorageDatabaseProvider.swift

```diff
@@ -0,0 +1,169 @@
+//
+//  SecureStorageDatabaseProvider.swift
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Common
+import GRDB
+
+public protocol SecureStorageDatabaseProvider {
```

Comment:
I think both of these can be removed from this protocol and pushed to the `makeDatabaseProvider`. I've got a [suggestion on a branch](https://github.com/duckduckgo/BrowserServicesKit/compare/brindy/secure-storage-suggestions..sam/secure-storage), but let's chat about it.